### PR TITLE
Improvement: Includes logic checks if project doesn't exist or no notifications found (#322)

### PR DESCRIPTION
**Pull request by CGoodwin90**
# Checklist
- [x] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated.
- [ ] Changelog entry has been written

Updates the existing project notifcation commands to utilize machinery, checks if a project is returned and whether notifications are found.

Requires https://github.com/uselagoon/machinery/pull/45

closes https://github.com/uselagoon/lagoon-cli/issues/322

### CODE REVIEW

File: cmd/notificationsemail.go [hunk 1]
```diff
@@ -154,26 +154,36 @@ var listProjectEmailsCmd = &cobra.Command{
...
 			debug)
-		result, err := lagoon.GetProjectNotificationEmail(context.TODO(), cmdProjectName, lc)
+
+		result, err := l.GetProjectNotificationEmail(context.TODO(), cmdProjectName, lc)
 		if err != nil {
 			return err
 		}
+		if len(result.Name) == 0 {
+			outputOptions.Error = fmt.Sprintf("No project found for '%s'\n", cmdProjectName)
+		} else if len(result.Notifications.Email) == 0 {
```
Comment (shreddedbacon) [inline cmd/notificationsemail.go:175]: this will panic still, you'll need to check for `nil` before you can inspect the result of `Notifications`

Comment (CGoodwin90) [reply cmd/notificationsemail.go:175]: I'm only seeing it panic if the project doesn't exist, which we're checking prior to inspecting `Notifications` so we never hit it. Unless I've missed something

Comment (shreddedbacon) [reply cmd/notificationsemail.go:175]: does it return `[]`?

Comment (CGoodwin90) [reply cmd/notificationsemail.go:175]: Yep! And that gets caught on line 175

Comment (shreddedbacon) [reply cmd/notificationsemail.go:175]: Great!

**Review by shreddedbacon — APPROVED** [review]

**Pull request merged**